test/AutoDiff/adjoint_expr_silgen.swift

```diff
@@ -0,0 +1,210 @@
+// RUN: %target-swift-frontend -emit-sil -verify %s | %FileCheck %s
+
+// Note: the adjoint calculations in this file are not intended to be accurate.
+
+//===----------------------------------------------------------------------===//
+// Top-level function
+//===----------------------------------------------------------------------===//
+
+@differentiable(reverse, adjoint: dSquare(_:primal:seed:))
+func square<T : FloatingPoint>(_ x: T) -> T {
+  return x * x
+}
+@_silgen_name("dsquare")
+private func dSquare<T : FloatingPoint>(_ x: T, primal: T, seed: T) -> T {
+  return 2 * x
+}
+
+@_silgen_name("test_top_level")
+func testTopLevel() -> Float {
+  let adjointSquare: (Float, Float, Float) -> Float = #adjoint(square)
+  return adjointSquare(1, 2, 3)
+}
+// CHECK-LABEL: sil hidden @test_top_level
+// CHECK: [[DSQUARE_RAW:%.*]] = function_ref @dsquare : $@convention(thin) <τ_0_0 where τ_0_0 : FloatingPoint> (@in_guaranteed τ_0_0, @in_guaranteed τ_0_0, @in_guaranteed τ_0_0) -> @out τ_0_0
+// CHECK: [[DSQUARE:%.*]] = partial_apply [callee_guaranteed] [[DSQUARE_RAW]]<Float>
```

Comment:
omit "[[DSQUARE:%.*]] = " to keep the test code simple?

same below

Reply:
I would rather keep it for clarity, if you don't mind.

Reply:
How does this help with clarify? I feel this is akin to defining a variable but not using it.

Reply:
It is the case that a variable is declared without being used, but the name of the variable provides clarity.

Reply:
I removed the unused variables in https://github.com/apple/swift/pull/17324/commits/4f13c8f533f8a990b1bda37ac2ad5e64810b2a9b. Thanks for the feedback!